src/library/searchqueryparser.cpp

```diff
@@ -104,51 +144,18 @@ QString SearchQueryParser::getTextArgument(QString argument,
             argument = tokens->takeFirst();
         }
     }
-
-    bool shouldMatchExactly = false;
+    StringMatch mode = StringMatch::Contains;
     if (argument.startsWith("=")) {
+        // strip the '=' from the argument
         argument = argument.mid(1);
-        shouldMatchExactly = true;
+        mode = StringMatch::Equals;
```

Comment:
Initially I wanted to make only super explicit `="string"` return exact/equal matches, this change extends it to `=string`.

Personally I'm fine with that, but it may be a bit confusing that `=string` does and `="string` does not find desired results.

wdyt?

Reply:
I think this is also the reason why the TextFilterNode tests fail now
https://github.com/ronso0/mixxx/actions/runs/6461369115/job/17540892333?pr=47

I'll now pick your commits, comment it and push. If that doesn't fix the tests we can revert it.

Reply:
Nope, the reason is that `=` is not considered below, hence `:"string"` is treated like `:="string"`.
I'll push a fixup.

Reply:
> Personally I'm fine with that, but it may be a bit confusing that =string does and ="string does not find desired results.

Well, I tried to achieve that `=` means I want to match exactly and `"..."` means I want to match multiple words. Those compose nicely and thus are easy to understand and apply. The only exception to this is if the string is not quoted completely. In this case I tried to emulate the behavior you implemented, even though I think we need to discuss more whether that makes sense. 

I can't quite wrap my head around why 
> =string does and ="string does not find desired results.

Can you give me an example?

Reply:
> > =string does and ="string does not find desired results.
>
> Can you give me an example?

On second thought, it's perfectly to search `contains` with unclosed quotes.
I reverted my fixup.